.edm4hep-ci.d/compile_and_test_with_spack.sh

```diff
@@ -0,0 +1,13 @@
+source .edm4hep-ci.d/setup_spack_gcc.sh
```

Comment:
Can we do this on the fly somehow, so that it doesn't depend on hardcoded paths in the setup script? I see you have tried to do it via `spack` loads, but opted for this setup script, why is that?

Reply:
Yes, I'm not sure what happens there, I think because maybe the docker image has no system g++, spack does not load gcc8.3 (which was compiled with the centos7 gcc). I can move this setup script to cvmfs.

Reply:
Ah yes, it kind of makes sense, that spack will not be able to locate the spack installed `gcc`, if it doesn't know with which compiler it has been built originally. We could potentially try to trick it into assuming that `gcc@4.8.5` is here by just filling the `.spack/linux/compilers.yaml` accordingly, but that is also brittle if the central spack installation changes.

In any case such a `setup_spack_gcc.sh` script might be nice to have in cvmfs, e.g. when one just needs a compiler and not the whole stack.